starter-ui/src/main/java/org/eclipse/starter/ui/StarterServlet.java

```diff
@@ -86,51 +86,62 @@ private synchronized Path generateProjectFromArchetype(Parameters p) throws IOEx
+
+        p.groupId = this.getParameterOrDefaultValue(req, "groupId", "com.example");
+        p.artifactId = this.getParameterOrDefaultValue(req, "artifactId", "demo");
+        p.version = this.getParameterOrDefaultValue(req, "version", "1.0.0-SNAPSHOT");
+        p.profile = this.getParameterOrDefaultValue(req, "profile", "api");
 
-        p.groupId = getParameterOrDefaultValue(req, "groupId", "com.example");
-        p.artifactId = getParameterOrDefaultValue(req, "artifactId", "demo");
-        p.version = getParameterOrDefaultValue(req, "version", "1.0.0-SNAPSHOT");
-        p.profile = getParameterOrDefaultValue(req, "profile", "api");
-        
         return p;
     }
 
-    private void requireNotNull(String archetypeGroupId, String archetypeArtifactId, String archetypeVersion) {
+    private void requireNotNull(final String archetypeGroupId, final String archetypeArtifactId, final String archetypeVersion) {
         Stream.of(new String[]{"archetypeGroupId", archetypeGroupId},
-                new String[]{"archetypeArtifactId", archetypeArtifactId},
-                new String[]{"archetypeVersion", archetypeVersion}
-        )
+                        new String[]{"archetypeArtifactId", archetypeArtifactId},
+                        new String[]{"archetypeVersion", archetypeVersion}
+                )
                 .filter(entry -> entry[1] == null)
                 .findAny()
                 .ifPresent(entry -> {
                     throw new RuntimeException("Query parameter " + entry[0] + " must be provided.");
                 });
     }
 
-    private String getParameterOrDefaultValue(HttpServletRequest req, String parameterName, String defaultValue) {
+    private String getParameterOrDefaultValue(final HttpServletRequest req, final String parameterName, final String defaultValue) {
         final String paramValue = req.getParameter(parameterName);
         return paramValue != null ? paramValue : defaultValue;
     }
 
-    private void zipDirectoryToOutput(Path generatedDirectory, ServletOutputStream outputStream) throws IOException {
-        try (ZipOutputStream zipOutStream = new ZipOutputStream(outputStream)) {
+    private void zipDirectoryToOutput(final Path generatedDirectory, final ServletOutputStream outputStream) throws IOException {
+        try (final ZipOutputStream zipOutStream = new ZipOutputStream(outputStream)) {
             new ZipCodec()
                     .addDirToZipArchive(generatedDirectory, zipOutStream);
         }
     }
 
-    private void deleteDirectoryRecursively(Path rootDirectory) throws IOException {
-        Files.walk(rootDirectory)
-                .sorted(Comparator.reverseOrder())
-                .map(Path::toFile)
-                .forEach(File::delete);
+    private void deleteDirectoryRecursively(final Path rootDirectory) throws IOException {
+        try (final var fileStream = Files.walk(rootDirectory)) {
+            //noinspection ResultOfMethodCallIgnored
```

Comment:
What does this mean?

Reply:
the method `File::delete` has a return that we ignore. Tools like Sonar (which I run on everything) fall over that. In this case, though it is intentional so I tell Sonar to ignore it.

Reply:
The try-with-resources refactor is just a bit safer programming.